### PR TITLE
Remove openapi.yml from build

### PR DESCRIPTION
**Pull request by teddyknox**

### CODE REVIEW

**Review by dib542 — COMMENTED** [review]
I will be glad to no longer see the line changes of the yml file on every PR 👏 

Shouldn't we also delete the actual yml file though?
- https://github.com/duality-labs/duality/blob/v0.2.2/docs/static/openapi.yml

And I think we can also delete these lines from the Ignite config:
- https://github.com/duality-labs/duality/blob/v0.2.2/config.yml#L23-L27

File: .circleci/config.yml [hunk 1]
```diff
@@ -106,14 +106,7 @@ jobs:
           command: mkdir -p /tmp/artifacts
       - run:
           name: Build chain binary
-          command: ignite chain build -o /tmp/artifacts --clear-cache
-      - run:
-          name: Ensure ignite has not generated changes which have not been commited to VC
```
Comment (jcompagni10) [inline .circleci/config.yml:111]: I think we still want this step since it also catches if there are changes to the .proto files that haven't been built.

**Review by jcompagni10 — DISMISSED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (teddyknox) [reply .circleci/config.yml:111]: maybe just do a proto-gen stage then

**Comment by teddyknox** [comment]
> I will be glad to no longer see the line changes of the yml file on every PR 👏
> 
> Shouldn't we also delete the actual yml file though?
> 
> * https://github.com/duality-labs/duality/blob/v0.2.2/docs/static/openapi.yml
> 
> And I think we can also delete these lines from the Ignite config:
> 
> * https://github.com/duality-labs/duality/blob/v0.2.2/config.yml#L23-L27

@dib542 do you think it's valid to leave it in, but not to check whether it's always up to date?

**Review by jcompagni10 — APPROVED** [review]

**Comment by dib542** [comment]
> > I will be glad to no longer see the line changes of the yml file on every PR 👏
> > Shouldn't we also delete the actual yml file though?
> > 
> > * https://github.com/duality-labs/duality/blob/v0.2.2/docs/static/openapi.yml
> > 
> > And I think we can also delete these lines from the Ignite config:
> > 
> > * https://github.com/duality-labs/duality/blob/v0.2.2/config.yml#L23-L27
> 
> @dib542 do you think it's valid to leave it in, but not to check whether it's always up to date?

If we plan on maintaining it regularly (at *least* on every release) then maybe we could leave it in the repo and have for example a separate CI task to update it when changes land in `main`. But if we don't plan to update it I'd rather have it gone. It's really frustrating to develop against an API spec to later find out that its wrong and several versions behind where it should be.

**Comment by teddyknox** [comment]
> > > I will be glad to no longer see the line changes of the yml file on every PR 👏
> > > Shouldn't we also delete the actual yml file though?
> > > 
> > > * https://github.com/duality-labs/duality/blob/v0.2.2/docs/static/openapi.yml
> > > 
> > > And I think we can also delete these lines from the Ignite config:
> > > 
> > > * https://github.com/duality-labs/duality/blob/v0.2.2/config.yml#L23-L27
> > 
> > 
> > @dib542 do you think it's valid to leave it in, but not to check whether it's always up to date?
> 
> If we plan on maintaining it regularly (at _least_ on every release) then maybe we could leave it in the repo and have for example a separate CI task to update it when changes land in `main`. But if we don't plan to update it I'd rather have it gone. It's really frustrating to develop against an API spec to later find out that its wrong and several versions behind where it should be.

If we could easily enforce that the openapi.yml if up to date we would. Since we can't it seems like overkill to stop generating the openapi.yml (and stop generating documentation from it) altogether.

**Review by dib542 — APPROVED** [review]
So I think I realized why I've been confused and why my comments have probably sounded confusing. I think we've been talking about completely different things. 

I thought the purpose of this PR was to remove `openapi.yml` from the repo so that it stopped generating non-deterministic unrelated code changes in every PR (and as a side-effect simplify the build checks because they'll be more deterministic). We had discussed previously if I needed `openapi.yml` for anything and I do not: for front-end purposes it can be removed from the repo.

I think now the original intended purpose of this PR may have been to remove an error in the build pipeline or something like that? to be fair, the PR has no purpose written in its description and is not linked to a GitHub issue, so its purpose is up for interpretation. 

Either way:
- I'm ok with removing it all for the purpose of cleaner PRs and easier build checks
- it's ok if you'd rather keep the docs in if you just want to fix a build step

About the concern about removing the docs and the generated documentation: I was able to create an interactive OpenAPI page locally at my http://localhost:1317 by adding the docs generation inside the Dockerfile, so creating the spec at build time. I added the changes needed at https://github.com/duality-labs/duality/commit/e8c65dc95cfd1f8e9948103e2ec3ba2c77be4c5c, and well it looks a bit hacky but it does work. I couldn't get the "github.com/ignite/cli/docs" package to serve the generated `openapi.yml`, maybe that's possible with some fiddling.

I think it is a valid concern that the development API page will disappear, we could resolve that here with more changes or keep the openapi.yml file around. It would be nice to have the page back for the next testnet.

**Comment by teddyknox** [comment]
> 

Ok, in that case we can remove the openapi.yml from the repo entirely. Updating to that effect.

**Review by jcompagni10 — APPROVED** [review]

**Pull request merged**